src/Linux/Host/Broker/Impl/Services/Utility.cs

```diff
@@ -0,0 +1,99 @@
+﻿
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
+using System.Text;
+using System.Threading;
+using Microsoft.Common.Core.OS;
+using Microsoft.Extensions.Logging;
+using Microsoft.R.Host.Broker.Startup;
+using Newtonsoft.Json;
+
+namespace Microsoft.R.Host.Broker.Services {
+    public class Utility {
+        public static bool AuthenticateUser(ILogger<IAuthenticationService> logger, IProcessServices ps,  string username, string password, out string profileDir) {
+            bool retval = false;
+            Process proc = null;
+            string userDir = string.Empty;
+            try {
+                proc = CreateRLaunchProcess(ps, true);
+                using (BinaryWriter writer = new BinaryWriter(proc.StandardInput.BaseStream))
+                using (BinaryReader reader = new BinaryReader(proc.StandardOutput.BaseStream)) {
+                    AuthenticationOnlyMessage message = new AuthenticationOnlyMessage();
+                    message.Username = username;
+                    message.Password = password;
+
+                    string json = JsonConvert.SerializeObject(message);
```

Comment:
The usual naming convention for fields in JSON is `camelCase` - can you enable that here?
```cs
SerializeObject(... ,new JsonSerializerSettings { 
        ContractResolver = new CamelCasePropertyNamesContractResolver() 
    })
```
(and adjust the C++ side accordingly)